Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -457,43 +563,62 @@ extension Channel {
-    ///         handshake.
-    ///   - streamConfiguration: The settings that will be used when establishing new streams. These mainly pertain to flow control.
+    ///   - connectionConfiguration: The settings that will be used when establishing the HTTP/2 connection.
+    ///         These will be sent to the peer as part of the handshake.
+    ///   - streamConfiguration: The settings that will be used when establishing new HTTP/2 streams.
+    ///         These mainly pertain to flow control.
     ///   - streamDelegate: The delegate to be notified in the event of stream creation and close.
-    ///   - h2ConnectionChannelConfigurator: An optional callback that will be invoked only
-    ///         when the negotiated protocol is H2 to configure the connection channel.
-    ///   - inboundStreamInitializer: A closure that will be called whenever the remote peer initiates a new stream.
-    /// - Returns: `EventLoopFuture<Void>` that completes when the channel is ready.
-    public func configureCommonHTTPServerPipeline(
+    ///   - http1ChannelConfigurator: An optional callback that will be invoked only
+    ///         when the negotiated protocol is HTTP/1.1 to configure the connection channel.
+    ///   - http2ChannelConfigurator: An optional callback that will be invoked only
+    ///         when the negotiated protocol is HTTP/2 to configure the connection channel.
+    ///   - http2StreamInitializer: A closure that will be called whenever the remote peer initiates a new stream.
+    /// - Returns: An `EventLoopFuture` containing a ``NIOTypedApplicationProtocolNegotiationHandler`` that completes when the channel
+    ///     is ready to negotiate. This can then be used to access the ``NIOProtocolNegotiationResult`` which may itself
+    ///     be waited on to retrieve the result of the negotiation
+    @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+    @_spi(AsyncChannel)
+    public func configureAsyncHTTPServerPipeline<HTTP1Output, HTTP2Output, HTTP2StreamOutput>(
         connectionConfiguration: NIOHTTP2Handler.ConnectionConfiguration,
         streamConfiguration: NIOHTTP2Handler.StreamConfiguration,
         streamDelegate: NIOHTTP2StreamDelegate? = nil,
-        h2ConnectionChannelConfigurator: ((Channel) -> EventLoopFuture<Void>)? = nil,
-        configurator: @escaping NIOHTTP2Handler.StreamInitializer
-    ) -> EventLoopFuture<Void> {
-        return self._commonHTTPServerPipeline(configurator: configurator, h2ConnectionChannelConfigurator: h2ConnectionChannelConfigurator) { channel in
-            channel.configureHTTP2Pipeline(
+        http1ChannelConfigurator: @escaping NIOChannelInitializerWithOutput<HTTP1Output>,
+        http2ChannelConfigurator: @escaping NIOChannelInitializerWithOutput<HTTP2Output>,
+        http2StreamInitializer: @escaping NIOChannelInitializerWithOutput<HTTP2StreamOutput>
```

Comment:
Should we add `inbound` in this parameters name?

Reply:
Configuration has been largely refactored.